### PR TITLE
DRAFT: [SDKIOS-905] Get rid of main thread warnings 

### PR DESCRIPTION
**Pull request by szymon-onewelcome**
- QRCodeViewController: fix for starting session on main thread + tiny refactor refactored

### CODE REVIEW

File: OneWelcomeExampleApp/ViewControllers/QR-code/QRCodeViewController.swift [hunk 1]
```diff
@@ -51,21 +51,68 @@ class QRCodeViewController: UIViewController {
...
-        captureSession = AVCaptureSession()
+        
+        setupCaptureSession()
+        startSession()
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
 
+        if captureSession?.isRunning == false {
```
Comment (kfa-onewelcome) [inline OneWelcomeExampleApp/ViewControllers/QR-code/QRCodeViewController.swift:62]: actually I don't like that ViewControllers holds everything (Massive View Controller)...
captusSession/preview Layer and its logic should be done in separate layers (interactor/presenter).

This would require a bit more work, but you started with the refactor already, you can continue it in proper way.

Comment (kfa-onewelcome) [reply OneWelcomeExampleApp/ViewControllers/QR-code/QRCodeViewController.swift:62]: if you don't know how to proceed, please create a separate story for the refactor and I will take it.

But I hope that after you catch the flow in oad-907, this should be doable for you

**Review by kfa-onewelcome — CHANGES_REQUESTED** [review]

File: OneWelcomeExampleApp/ViewControllers/QR-code/QRCodeViewController.swift (hunk 1 above)
Comment (szymon-onewelcome) [reply OneWelcomeExampleApp/ViewControllers/QR-code/QRCodeViewController.swift:62]: Actually the aim was not to refactor but to fix main thread issue. So lets merge this one and create a separate story for refactor.

Comment (szymon-onewelcome) [reply OneWelcomeExampleApp/ViewControllers/QR-code/QRCodeViewController.swift:62]: TLDR please resolve the comment and approve if you don't have any other remarks.

Comment (kfa-onewelcome) [reply OneWelcomeExampleApp/ViewControllers/QR-code/QRCodeViewController.swift:62]: ok

**Review by kfa-onewelcome — APPROVED** [review]
